contracts/V1/StateChannelDiamondProxy/FraudProofFacet.sol

```diff
@@ -151,6 +152,28 @@ contract FraudProofFacet is StateChannelCommon {
         
         return signer;
     }
+
+    function _handleBlockInvalidPreviousLink(bytes memory encodedProof, FraudProofVerificationContext memory fraudProofVerificationContext) internal pure returns (address) {
+        BlockInvalidPreviousLinkProof memory blockInvalidPreviousLinkProof = abi.decode(encodedProof, (BlockInvalidPreviousLinkProof));
+        Block memory fraudBlock = abi.decode(blockInvalidPreviousLinkProof.invalidBlock.encodedBlock, (Block));
+                  
+        if(fraudBlock.transaction.header.transactionCnt == 0){
+            bytes memory previousStateMachineState = blockInvalidPreviousLinkProof.previousStateMachineState;
+            require(fraudBlock.previousBlockHash != keccak256(abi.encode(previousStateMachineState)), ErrorValidPreviousLink());
+        }else{
+
+            Block memory previousBlock = abi.decode(blockInvalidPreviousLinkProof.previousBlock.encodedBlock, (Block));
+            require(previousBlock.transaction.header.channelId == fraudBlock.transaction.header.channelId
+            && fraudBlock.transaction.header.channelId == fraudProofVerificationContext.channelId, ErrorNotSameChannelId()
+            );
+            require(fraudBlock.previousBlockHash != keccak256(abi.encode(previousBlock)), ErrorValidPreviousLink());
+        }
+        address signer = StateChannelUtilLibrary.retriveSignerAddress(
+            blockInvalidPreviousLinkProof.invalidBlock.encodedBlock,
+            blockInvalidPreviousLinkProof.invalidBlock.signature
+        );
+        return signer;
+    }
```

Comment:
@MrishoLukamba please explain what you wanted to do with this fraud proof and where it's needed? The logic here isn't correct (blockHeight 0 previousBlockHash is hash(genesisStateSnapshot)) + you never check are blockHeights sequential 

All of that aside this can not be a fraud proof, since the guy producing the 'not linked block' can be on a different fork due to someone else performing a double sign - also don't understand where this is needed